### ISSUE TITLE
Timer callbacks fail if the server is busy

### ISSUE THREAD

**Issue by Tokiko1** [tgstation/tgstation#29314]
Sorry in advance if this is vague or if this happens to be a fluke, but this was a strange behaviour that I was able to consistently reproduce while trying to make a lasergun that freezes stuff in the way freon does.

Basically, I noticed that the server will skip/fail/forget timed callbacks when it's very busy. Busy here being locked up by processing stuff so that even movement is not possible but this may vary depending on how good/bad your computer/server is.

The only confirmed bug that this causes are the beam overlays that never get deleted, but I am confident that this might be responsible for various other bugs, like crashes during freon floods, global timers breaking and the mc crashes.

Causes #24475, #28626, possibly others.

**Comment by Cyberboss** [comment]
Probably related to #23818

**Comment by MrStonedOne** [comment]
I wasn't able to reproduce.

**Comment by Tokiko1** [comment]
Here is how I caused this bug, perhaps you will be able to reproduce it this way:

I made a freezing gun that draws a beam using the atom.Beam() with a short duration and then does some rather costly icon operations(old freon freeze overlays) on a whole bunch of objects. Server freezes for a moment, beam fails to be deleted with its timed callback. I replaced the timed callback with a simple sleep() and the beam got correctly deleted, despite the freeze.
